lib/Dialect/RTL/RTLOps.cpp

```diff
@@ -713,6 +713,56 @@ void ArrayCreateOp::build(OpBuilder &b, OperationState &state,
   build(b, state, ArrayType::get(elemType, values.size()), values);
 }
 
+static ParseResult parseArrayConcatTypes(OpAsmParser &p,
```

Comment:
IMO, this is fancy but actually makes it harder to read the generated IR.  I'd recommend just going with:
`%38 = rtl.array_concat %subArr, %subArr : !rtl.array<3xi4>, !rtl.array<3xi4>`

instead of:
`%38 = rtl.array_concat %subArr, %subArr : (3, 3 x i4)`.

Reply:
Yeah, I just worry about legibility when you're concatenating a _lot_ of operands. (Which is standard with ESI ;) )

Reply:
k, I think it is still best to go with the simple version of this.  MLIR text unavoidably has redundant types everywhere.  Having something simple when you say "call op->dump()" in a debugger is a good thing.

Reply:
True, and if you're concat'ing a bunch of operands, the operand list is gonna kill you anyway.

Reply:
right